paimon-core/src/main/java/org/apache/paimon/table/source/TableScan.java

```diff
@@ -47,4 +47,7 @@ public interface TableScan {
     interface Plan {
         List<Split> splits();
     }
+
+    /** Close this scan, clean resources here. */
+    void close();
```

Comment:
`TableScan` is a `@Public` interface, do we really need to add a `close()` method here?